spec/models/prison_spec.rb

```diff
@@ -0,0 +1,166 @@
+require 'rails_helper'
+
+RSpec.describe Prison, type: :model do
+  let(:mock_slots_data) do
+    {
+      "mon"=>["1400-1600"],
+      "tue"=>["1400-1600"],
+      "wed"=>["1400-1600"],
+      "thu"=>["1400-1600"],
+      "fri"=>["1400-1600"],
+      "sat"=>["0930-1130", "1400-1600"],
+      "sun"=>["1400-1600"]
+    }
+  end
+
+  let(:mock_prison_data) do
+    {
+      "nomis_id"=>"RCI",
+      "canned_responses"=>true,
+      "enabled"=>true,
+      "phone"=>"01634 803100",
+      "email"=>"pvb.rochester@maildrop.dsd.io",
+      "instant_booking"=>false,
+      "address"=>["1 Fort Road", "Rochester", "Kent", "ME1 3QS"],
+      "lead_days"=>4,
+      "booking_window"=>14,
+      "works_weekends"=>true,
+      "unbookable"=>[Date.new(2015, 7, 29), Date.new(2015, 12, 25)],
+      "slot_anomalies"=>{Date.new(2015, 8, 14)=>["0700-0900"]},
+      "slots"=> mock_slots_data
+    }
+  end
+
+  describe '#find' do
+    context 'when a prison exists' do
+      it 'returns a prison instance' do
+        expect(described_class::PRISONS_DATA).to receive(:[]).
+          with('Example Prison') { mock_prison_data }
+
+        prison = described_class.find 'Example Prison'
+        expect(prison).to be_kind_of described_class
+      end
+    end
+    context 'when a prison does not exist' do
+      it 'raises a PrisonNotFound error' do
+        allow(described_class::PRISONS_DATA).to receive(:[]) { nil }
+
+        expect { described_class.find('nothing') }.
+          to raise_error(described_class::PrisonNotFound)
+      end
+    end
+  end
+
+  def constructor_for(prison_data)
+    described_class.new 'Example Prison', prison_data
+  end
+
+  subject { constructor_for mock_prison_data }
+
+  describe '.name' do
+    it 'returns the name of the prison' do
+      expect(subject.name).to eq 'Example Prison'
+    end
+  end
+
+  describe '.unbookable_dates' do
+    context 'when a prison has unbookable dates' do
+      let(:expected_dates) { Set.new [Date.new(2015, 7, 29), Date.new(2015, 12, 25)] }
+
+      it 'returns a set containing the dates' do
+        expect(subject.unbookable_dates).to eq expected_dates
+      end
+    end
+    context 'when a prison does not have unbookable dates' do
+      subject { constructor_for mock_prison_data.except 'unbookable' }
+
+      it 'returns an empty set' do
+        subject.unbookable_dates.tap do |set|
+          expect(set).to be_empty
+          expect(set).to be_kind_of Set
+        end
+      end
+    end
+  end
+
+  describe '.visiting_slots' do
+    it 'returns the prisons slots as a hash' do
+      expect(subject.visiting_slots).to eq mock_slots_data
+    end
+  end
+
+  describe '.visiting_slot_days' do
+    let(:expected_slot_days) { %w<mon tue wed thu fri sat sun> }
+    it 'returns the abbreviated day names of available visiting days' do
+      expect(subject.visiting_slot_days).to eq expected_slot_days
+    end
```

Comment:
I wonder if it's worth testing that this works with another set of visiting slot days?